### PR TITLE
refactor!: remove role privilege rules

### PR DESCRIPTION
**Pull request by TylerGillson**
## Issue
N/A

## Description
- Remove RolePrivilegeRules
- Add support for privilege rules on all entities
- Configure `GroupPrincipals` and `Propagated` for privilege rules
- Bump golangci-lint and fix import issues related to `_validator`

Requires:
- https://github.com/validator-labs/validator-plugin-vsphere/pull/362

### CODE REVIEW

**Review by ahmad-ibra — APPROVED** [review]

File: pkg/services/clouds/aws_service.go
[the remark this answers is not in the document]
Comment (ahmad-ibra) [reply pkg/services/clouds/aws_service.go:1]: time to rename my creds file 🙌

File: pkg/services/validator/vmware.go [hunk 1]
```diff
@@ -518,59 +408,93 @@ func configureEntityPrivilegeRules(ctx context.Context, c *components.VspherePlu
...
+	}
+
+	log.InfoCLI(`
+	Is the permission that assigns privileges to the user for the selected entity expected to
+	have propagation enabled? If multiple permissions for group principals associated with the
+	user exist on the entity in question, propagation will be consider enabled if any of them
+	has propagation enabled. If a permission exists for the entity whose principal is the user,
+	the propagation value of that permission takes precedence.
+	`)
+	r.Propagation.Propagated, err = prompts.ReadBool("Propagated", false)
```
Comment (ahmad-ibra) [inline pkg/services/validator/vmware.go:495]: should this only be prompted for if `propagation validation` is enabled above? ie if propogation validation is _not_ enabled, then users probably wouldn't care if an entity is expected to have propagation enabled.

File: pkg/services/validator/vmware.go [hunk 2]
```diff
@@ -825,98 +760,107 @@ func getClusterName(ctx context.Context, datacenter string, driver vsphere.Drive
...
 func getEntityAndClusterInfo(ctx context.Context, entityType string, driver vsphere.Driver, datacenter string) (entityName, clusterName string, err error) {
-	switch entityType {
-	case cfg.ValidatorVsphereEntityCluster, "cluster":
+	switch e := entity.Map[entityType]; e {
+	case entity.Cluster:
 		entityName, err = getClusterName(ctx, datacenter, driver)
 		if err != nil {
 			return "", "", err
 		}
 		return entityName, entityName, nil
```
[the remark this answers is not in the document]
Comment (ahmad-ibra) [reply pkg/services/validator/vmware.go:770]: disregard, the entity is a cluster in this case 🤦

**Review by ahmad-ibra — APPROVED** [review]
Left a question which _may_ result in changes, but everything looks good to me

File: pkg/services/validator/vmware.go (hunk 1 above)
Comment (TylerGillson) [reply pkg/services/validator/vmware.go:495]: There's an early return for that.

**Pull request merged**